Sources/Controller/ImagePickerController+Assets.swift

```diff
@@ -31,6 +31,11 @@ extension ImagePickerController: AssetsViewControllerDelegate {
                 imagePickerDelegate?.imagePicker(self, didDeselectAsset: first)
             }
         }
+        else if settings.selection.notifyReachingMaxLimit && assetStore.count > settings.selection.max {
```

Comment:
Changed else if  to  if.
Remove and unselect has to be there.Because-
Suppose maxselection is 2
When didSelectAsset gets called  when we are on 3rd selection, then only this loop will get executed.so that asset has to be removed.If we are not removing that asset.then we will have 3 assets selected instead of 2 assets.